src/psyclone/tests/psyir/nodes/codeblock_test.py

```diff
@@ -120,18 +121,30 @@ def test_codeblock_get_symbol_names(parser):
       END DO myloop
     end subroutine mytest''')
     prog = parser(reader)
-    block = CodeBlock(prog.children, CodeBlock.Structure.STATEMENT)
+    scope = Schedule()
+    block = CodeBlock(prog.children, CodeBlock.Structure.STATEMENT,
+                      parent=scope)
     sym_names = block.get_symbol_names()
-    assert "a" in sym_names
-    assert "b" in sym_names
-    assert "c" in sym_names
-    assert "mytest" in sym_names
-    assert "subroutine" not in sym_names
-    assert "sqrt" not in sym_names
-    assert "myloop" not in sym_names
-    assert "myifblock" not in sym_names
-    assert "this_is_true" in sym_names
-    assert "that_is_true" in sym_names
+    refs = block.walk(Reference)
```

Comment:
Please could you also check that these refs are immediate children of the CodeBlock (as this is new).

Your use of 'virtual refs' to describe them did make me wonder whether we should have a new `VirtualReference` subclass of  `Reference` but I'm not sure that buys us anything other than making it clear that these aren't "real" references.